2.6/root/usr/share/container-scripts/mongodb/init-functions.sh

```diff
@@ -0,0 +1,83 @@
+# mongo_create_admin creates the MongoDB admin user with password: MONGODB_ADMIN_PASSWORD
+# $1 - login parameters for mongo (optional)
+# $2 - host where to connect (localhost by default)
+function mongo_create_admin() {
+  if [[ -z "${MONGODB_ADMIN_PASSWORD:-}" ]]; then
+    echo "=> MONGODB_ADMIN_PASSWORD is not set. Authentication can not be set up."
+    exit 1
+  fi
+
+  # Set admin password
+  set +e
+  mongo admin ${1:-} --host ${2:-"localhost"} --eval "db.createUser({user: 'admin', pwd: '${MONGODB_ADMIN_PASSWORD}', roles: ['dbAdminAnyDatabase', 'userAdminAnyDatabase' , 'readWriteAnyDatabase','clusterAdmin' ]});"
+  local result=$?
+  set -e
```

Comment:
Because mongo call could fail to connect to mongod. So to be able to check it and to print error message.

Is it too complicated? Is mongo fail message enought?